client-v2/src/components/FrontsCAPIInterface/SearchInput.tsx

```diff
@@ -1,124 +1,130 @@
+import CAPITagInput, { SearchTypes } from '../FrontsCAPIInterface/TagInput';
+import CAPIFieldFilter from '../FrontsCAPIInterface/FieldFilter';
 import CAPIDateRangeInput from '../FrontsCAPIInterface/DateInput';
 import { getIdFromURL } from 'util/CAPIUtils';
 import { getTodayDate } from 'util/getTodayDate';
 import moment from 'moment';
+import { fetchLive, fetchPreview } from 'bundles/capiFeedBundle';
+import { Dispatch } from 'types/Store';
+import { connect } from 'react-redux';
+import FilterItem from './FilterItem';
+import debounce from 'lodash/debounce';
 
 interface FrontsCAPISearchInputProps {
   children: any;
   additionalFixedContent?: React.ComponentType<any>;
   displaySearchFilters: boolean;
   updateDisplaySearchFilters: (value: boolean) => void;
   isPreview: boolean;
-}
-
-type SearchTypeMap<T> = { [K in SearchTypes | FilterTypes]: T };
-
-type SearchTerms = SearchTypeMap<string>;
-type SelectedTags = SearchTypeMap<string[]>;
-
-interface FrontsCAPISearchInputState {
-  q: string | void;
-  // Was the current search string derived from a URL?
-  isResource: boolean;
-  searchTerms: SearchTerms;
-  selected: SelectedTags;
-  fromDate: moment.Moment | null;
-  toDate: moment.Moment | null;
+  fetchFeed: (q: string, liveParams: object, previewParams: object) => void;
 }
 
 const InputContainer = styled('div')`
   margin-bottom: 20px;
   background: #ffffff;
 `;
 
-const SearchTermItem = styled('div')`
-  color: #121212;
-  font-weight: bold;
-  border: solid 1px #c4c4c4;
-  font-size: 14px;
-  background-color: #ffffff;
-  padding: 7px 15px 7px 15px;
-  margin-bottom: 10px;
-  :hover {
-    color: #c4c4c4;
-  }
-`;
+interface StringArrSearchItems {
+  tags: string[];
+  sections: string[];
+  desks: string[];
+  ratings: string[];
+}
+
+type FrontsCAPISearchInputState = StringArrSearchItems & {
+  query: string;
+  fromDate: null | moment.Moment;
+  toDate: null | moment.Moment;
+};
 
-const emptySearchTerms = {
-  tags: '',
-  sections: '',
-  desks: '',
-  ratings: '',
-  fromDate: null,
-  toDate: null
+const getParams = (
+  {
+    tags,
+    sections,
+    desks,
+    ratings,
+    query,
+    toDate: to,
+    fromDate: from
+  }: FrontsCAPISearchInputState,
+  isPreview: boolean
+) => ({
+  q: query,
+  tag: [...tags, ...desks].join(','),
+  section: sections.join(','),
+  'star-rating': ratings.join('|'),
+  'from-date': from && from.format('YYYY-MM-DD'),
+  'to-date': to && to.format('YYYY-MM-DD'),
+  'page-size': '20',
+  'show-elements': 'image',
+  'show-fields': 'internalPageCode,trailText,firstPublicationDate,isLive',
+  ...(isPreview
+    ? { 'order-by': 'oldest', 'from-date': getTodayDate() }
+    : { 'order-by': 'newest', 'order-date': 'first-publication' })
+});
+
+const renderDateAsString = (date: moment.Moment | null) => {
+  if (!date) {
+    return 'Not selected';
+  }
+  return date.format('DD/MM/YYYY');
 };
 
-const emptyState = {
-  q: undefined,
-  isResource: false,
-  searchTerms: emptySearchTerms,
```

Comment:
I took search terms out from the parent and left them in their respective child components just to thin this component out a bit. Potentially could be reverted to how it was but they weren't currently being used so it was just another thing to handle here.